[AutoInstall] Allow wheels by default

gsnedders opened the pull request:
#### dba0adf164394251846b0da1769c06de85097666
<pre>
[AutoInstall] Allow wheels by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=263119">https://bugs.webkit.org/show_bug.cgi?id=263119</a>

Reviewed by Jonathan Bedard.

This allows packages to be installed by wheels by default; if wheel=True
is passed then only wheels will be accepted, if wheel=False is passed
then wheels will never be accepted.

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(Package.__init__):
(Package.archives):
(AutoInstall):
(AutoInstall.enabled):
(AutoInstall.temporarily_disable):
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py:
* Tools/Scripts/webkitpy/autoinstalled/twisted.py:

Canonical link: <a href="https://commits.webkit.org/269817@main">https://commits.webkit.org/269817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a2701e864931452c1bacf2acb5a193fa53052e6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23645 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1761 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24766 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25804 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21828 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23914 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3341 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24181 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22389 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23889 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1313 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20458 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26400 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/23828 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1101 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21364 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27642 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21569 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21633 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25405 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1073 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18776 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/23438 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1052 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5658 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1486 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1363 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
```diff
@@ -474,10 +492,19 @@ def _request(cls, url, ca_cert_path=None):
         return True if cls.directory else None
 
+    @classmethod
+    @contextmanager
```
Why is this a context manager? Seems like `classmethod` would suffice.

gsnedders replied:
See the `yield` statement. We want to be able to temporarily disable it, and then re-enable it afterwards.

JonWBedard replied:
Ah! Sorry, missed that. Was thinking "memoized" for some reason. Yes, that makes sense.

JonWBedard reviewed (commented):
Generally in favor of this, as @gsnedders and I have discussed a number of times. I'd like to do some local validation with some `webkitcorepy` clients that are not tested in EWS before r+ing.

JonWBedard reviewed (approved):
Validated this with a number of internal scripts as well.

gsnedders committed:
[AutoInstall] Allow wheels by default
https://bugs.webkit.org/show_bug.cgi?id=263119

Reviewed by Jonathan Bedard.

This allows packages to be installed by wheels by default; if wheel=True
is passed then only wheels will be accepted, if wheel=False is passed
then wheels will never be accepted.

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(Package.__init__):
(Package.archives):
(AutoInstall):
(AutoInstall.enabled):
(AutoInstall.temporarily_disable):
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py:
* Tools/Scripts/webkitpy/autoinstalled/twisted.py:

Canonical link: https://commits.webkit.org/269817@main
